packages/core/src/login/webview/vue/toolkit/backend_toolkit.ts

```diff
@@ -75,14 +75,54 @@ export class ToolkitLoginWebview extends CommonAuthWebview {
     async errorNotification(e: AuthError) {
         await vscode.window.showInformationMessage(`${e.text}`)
     }
-
+    /**
+     * Returns list of connections that are pushed from Amazon Q to Toolkit
+     */
     async fetchConnections(): Promise<AwsConnection[] | undefined> {
-        //This does not need to be implement in aws toolkit vue backend
-        return undefined
+        const connections: AwsConnection[] = []
+        const _connections = await Auth.instance.listConnections()
+        _connections.forEach(c => {
+            const status = Auth.instance.getConnectionState({ id: c.id })
+            const source = Auth.instance.getConnectionSource({ id: c.id })
+            if (c.type === 'sso' && source === 'amazonq' && status) {
+                connections.push({
+                    id: c.id,
+                    label: c.label,
+                    type: c.type,
+                    ssoRegion: c.ssoRegion,
+                    startUrl: c.startUrl,
+                    state: status,
+                } as AwsConnection)
+            }
+        })
+        return connections
     }
-
+    /**
+     * Re-use connection that is pushed from Amazon Q to Toolkit.
+     */
     async useConnection(connectionId: string): Promise<AuthError | undefined> {
-        return undefined
+        return this.ssoSetup('useConnection', async () => {
+            const conn = await Auth.instance.getConnection({ id: connectionId })
+            if (conn === undefined || conn.type !== 'sso') {
+                return
+            }
+            if (this.isCodeCatalystLogin) {
+                if (conn.scopes?.includes(scopesCodeCatalyst[0])) {
```

Comment:
- This codecatalyst code is a better fit in the class `CodeCatalystAuthenticationProvider`. The methods `connectToAwsBuilderID()` and `connectToEnterpriseSso()` are very similar to what you are doing, so you probably just need to do some refactorings to create a `useConnection()` method in `CodeCatalystAuthenticationProvider` that reuses some of their code
- The function `isValidCodeCatalystConnection()` is also the correct way to determine if the connection is usable, you can see this used in the methods I mentioned above.

Reply:
>This codecatalyst code is a better fit in the class CodeCatalystAuthenticationProvider

Not all of these code. Most code are specific to the new login webview and it contains cross extension interaction that should only exists in the webview backend code. These code utilizes CodeCatalystAuthenticationProvider to login to CC however it also updates the states of the login webview. CodeCatalystAuthenticationProvider should never be able to update login webview states. 


>The function isValidCodeCatalystConnection() is also the correct way to determine if the connection is usable, you can see this used in the methods I mentioned above.

These are exactly the same as `(conn.scopes?.includes(scopesCodeCatalyst[0]))` since the scopes only has one element in the list. I can make the change in case future scopes changes.